### PR TITLE
Yql

### PR DESCRIPTION
**Pull request by dianpan**
@arf563

### CODE REVIEW

File: app/models/fetcher.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+require 'json'
+require 'net/http'
+
+def get_yql_data(ticker, startDate, endDate)
```
Comment (arf563) [inline app/models/fetcher.rb]: Please add comments describing the expected format of the arguments.  E.g.:

// Params:
// ticker a string representing the stock ticker symbol
// startDate and endDate together represent the range of dates for which data should be fetched.
//   startDate is required, endDate may be empty.  If endDate is empty, the current system time will be used.
//   startDate and endDate are both strings in the MM-DD-YYYY format.
// Returns an array of result objects with the following format:
// ...

Fix up the description of endDate though-- how are "optional" parameters usually handled in ruby?  would the caller leave it out?  pass a null / nil value?  an empty string?

File: app/models/fetcher.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  url = "https://query.yahooapis.com/v1/public/yql?q=select%20*%20from%20yahoo.finance.historicaldata%20where%20symbol%20%3D%20%22#{ticker}%22%20and%20startDate%20%3D%20%22#{startDate}%22%20and%20endDate%20%3D%20%22#{endDate}%22&format=json&env=store%3A%2F%2Fdatatables.org%2Falltableswithkeys&callback="
+  resp = Net::HTTP.get_response(URI.parse(url))
+  result = JSON.parse(resp.body)
+  return result
+end
+
+puts JSON.pretty_generate(get_yql_data("YHOO", "2015-01-01", "2015-08-26"))
```
Comment (arf563) [inline app/models/fetcher.rb]: assume this is for testing?  Move to test file?

File: Gemfile [hunk 3]
```diff
@@ -30,10 +30,13 @@ gem 'sdoc', '~> 0.4.0', group: :doc
 # Use Capistrano for deployment
 # gem 'capistrano-rails', group: :development
 gem 'composite_primary_keys'
+gem 'nas-yahoo_stock'
```
Comment (arf563) [inline Gemfile]: afaict, you don't actually use either of these gems?  You are making direct, hand rolled webservice calls to yql using net libraries, so you could probably remove both of these?

File: app/models/fetcher.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
```
Comment (arf563) [inline app/models/fetcher.rb]: Nit: remove blank lines

**Comment by dianpan** [comment]
@arf563 
seeded the stock table with a list of all stock symbols
make sure you run "rake db:seed"

**Pull request merged**